File: js-src/forms/form_radios.js
/**
 * Extraction utilities for radio button controls.
 */
const { findLabelFor, findGroupLabel, getAssociatedText } = require('./form_common');

/**
 * Extract radio button groups
 * @param {Object} $ - Cheerio instance
 * @param {Object} container - Container element
 * @returns {Array} Array of radio group objects
 */
function extractRadioGroups($, container) {
  const radioGroups = {};
  
  $(container).find('input[type="radio"]').each((i, radio) => {
    const name = $(radio).attr('name');
    if (!name) {
      return;
    }
    
    if (!radioGroups[name]) {
      radioGroups[name] = {
        name: name,
        options: [],
        label: findGroupLabel($, name)
      };
    }
    
    const option = {
      id: $(radio).attr('id') || '',
      value: $(radio).attr('value') || '',
      checked: $(radio).attr('checked') !== undefined,
      disabled: $(radio).attr('disabled') !== undefined
    };
    
    // Try to find label for this specific radio button
    const optionLabel = findLabelFor($, radio);
    if (optionLabel) {
      option.label = optionLabel;
    }
    
    // Get associated text (might be different from label)
    const optionText = getAssociatedText($, radio);
    if (optionText && optionText !== option.label) {
      option.text = optionText;
    }
    
    radioGroups[name].options.push(option);
  });
  
  return Object.values(radioGroups);
}

module.exports = { extractRadioGroups };

File: src/forms/form_radios.py
"""
Extraction utilities for radio button controls.
"""
from forms.form_common import find_label_for, find_group_label, get_associated_text

def extract_radio_groups(container):
    """Extract radio button groups"""
    radio_groups = {}
    radios = container.find_all('input', type='radio')
    
    for radio in radios:
        name = radio.get('name', '')
        if not name:
            continue
            
        if name not in radio_groups:
            radio_groups[name] = {
                'name': name,
                'options': [],
                'label': find_group_label(container, name)
            }
        
        option = {
            'id': radio.get('id', ''),
            'value': radio.get('value', ''),
            'checked': radio.has_attr('checked'),
            'disabled': radio.has_attr('disabled')
        }
        
        # Try to find label for this specific radio button
        option_label = find_label_for(container, radio)
        if option_label:
            option['label'] = option_label
            
        # Get associated text (might be different from label)
        option_text = get_associated_text(container, radio)
        if option_text and option_text != option.get('label', ''):
            option['text'] = option_text
            
        radio_groups[name]['options'].append(option)
    
    return list(radio_groups.values())
